src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Takes in user inputs and make sense of them.
+ */
+public class Parser {
+    /**
+     * Parses the inputs and calls the corresponding execution.
+     *
+     * @param tasklist The overall user's task list.
+     */
+    public static void parseUserInput(TaskList tasklist) {
```

Comment:
I think the parameter can be renamed to taskList (in camelcase).

Reply:
Thank you! Yes, seems like I made quite a number of careless mistakes 😅